### DISCUSSION TITLE
API issues?

### DISCUSSION THREAD

**Discussion by viking2010** [hultenvp/solis-sensor#280] [General]
Anyone else having issues with data being pulled in to HA this morning (22nd April). Around 9am I spotted that instead of updates being entered every couple of mins, it was only updating every 10 minutes. So I restarted HA and the underlying OS, and now it's not picking anything up and says unavailable. The Solis app is still showing regular updates and the WiFi data logger is responding as it should. Having tried the API test script, it worked once but then I'm now getting errors (See below). I suspect this is something Solis are messing with at their end. They're a bit of a law unto themselves! Be nice if the gave a warning about any maintenance. 

Test script Python Error:
![image](https://user-images.githubusercontent.com/100141279/233776687-374d8ac1-4523-4078-af75-88dc8e708016.png)

**Comment by viking2010** [comment]
From what I can ascertain from various forums, it would appear the API is broken at Solis' end.

**Comment by curry684** [comment]
Not sure what the source is but there's a typo in your call - the correct method name is `/v1/api/inverterDetail`. So the call in your screenshot is expected to fail consistently.

![image](https://user-images.githubusercontent.com/1455673/234307886-d7476580-0e4f-4992-9a85-f5e92db65ff9.png)

**Comment by LucidityCrash** [comment]
You've not been around here long :) ... there was a typo in the API originally (probably more than one)  and inveterDetail does infact work.  That isn't to say inverterDetail doesn't work now - it didn't in the past.
I've just got inveterDetail to work but it is now back to 502's again ... inverterDetail has only responded with 502's for me so far.

**Reply by LucidityCrash** [reply]
update ... inverterDetail has just returned a proper result (so did inveterDetail) :)

**Reply by viking2010** [reply]
not the only typo in that document either! I think there was one for Bsttery instead of battery....or something like that!  I'm still only getting API data sporadically. I suspect some of the issues could be the transfer of user data from the old to the new portal. It will no doubt be having a knock on effect on their systems. As ever, would have been nice for them to let folks know! :-D

**Reply by curry684** [reply]
Heh funny, indeed started using it last week.

**Reply by viking2010** [reply]
@LucidityCrash when you run the script do you get the same errors I was getting? Then sometimes the script will produce output. I've raised a ticket with Solis but they said "maybe it's an issue with your API tools".  I am in the process of replying to say, well if it stopped working completely, then yes I could understand that, but it's working periodically.  So if the script was faulty, then it wouldn't work at all....ever!

Maybe we all need to log some tickets to Solis about this!

**Reply by LucidityCrash** [reply]
Yes I get the same error, and then sometimes it works ... which is exactly what happens in the plugin in HA.
I modified the script a little and changed the final print line and added a print(x.text)  this gives the raw output of the request guess what it says ???

```
<html>
<head><title>502 Bad Gateway</title></head>
<body bgcolor="white">
<center><h1>502 Bad Gateway</h1></center>
<hr><center>nginx</center>
</body>
</html>
```

errr ... that would be a server issue : 
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/502

**Reply by viking2010** [reply]
> errr ... that would be a server issue :
> https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/502

lol, it sure would!! Glad you get the same issue. Just helps to confirm I'm not doing something daft at my end! Thank you!

**Comment by Davel-GH** [comment]
I have also started getting 502 with occasional 504 messages from the solis API. I ping it every 6mins and now only get a successful response 1 in 10 times. I have raised a ticket with solis.

**Comment by dpgh947** [comment]
The constant "no inverters found" errors in my logs stopped at 06:39 this morning (over 2 hours ago now), and the test script is also working every time now, the first time I have seen it work. I think (hope) they have fixed it.

**Comment by viking2010** [comment]
Yes I've had the same experience this morning too with the data production. Bang on every 2 mins there's an update now. I was lucky quite early on with Solis to be able to get 1 minute updates, but I set the integration to check every 2 mins. Every 1 minute doesn't always work reliably, but 2 mins is rock solid usually.
